griptape/drivers/prompt/huggingface_pipeline_prompt_driver.py

```diff
@@ -34,28 +35,38 @@ class HuggingFacePipelinePromptDriver(BasePromptDriver):
         ),
         kw_only=True,
     )
+    pipe: TextGenerationPipeline = field(
+        default=Factory(
+            lambda self: import_optional_dependency("transformers").pipeline(
+                "text-generation", model=self.model, max_new_tokens=self.max_tokens, tokenizer=self.tokenizer.tokenizer
+            ),
+            takes_self=True,
+        )
+    )
 
     def try_run(self, prompt_stack: PromptStack) -> TextArtifact:
-        prompt = self.prompt_stack_to_string(prompt_stack)
-        pipeline = import_optional_dependency("transformers").pipeline
+        messages = [{"role": input.role, "content": input.content} for input in prompt_stack.inputs]
 
-        generator = pipeline(
-            tokenizer=self.tokenizer.tokenizer,
-            model=self.model,
-            max_new_tokens=self.tokenizer.count_output_tokens_left(prompt),
+        result = self.pipe(
+            messages,
+            max_new_tokens=self.max_tokens,
+            eos_token_id=[
+                self.tokenizer.tokenizer.eos_token_id,
+                *[self.pipe.tokenizer.convert_tokens_to_ids(token) for token in self.tokenizer.stop_sequences],
+            ],
```

Comment:
I'm having trouble finding the docs related to this one, could you share your sources?

I would have expect `eos_token_id` to be set automatically, since the pipeline has access to the model and tokenizer. The one case where I'd expect this to be required is for model specific differences. Did you test this will multiple models?

> Note that I did not see `eos_token_id` specified in the hugging face example usage of [TinyLlama/TinyLlama-1.1B-Chat-v1.0](https://huggingface.co/TinyLlama/TinyLlama-1.1B-Chat-v1.0)

Reply:
I wasn't too sure about this TBH. I got it from [this article](https://huggingface.co/blog/llama3#using-%F0%9F%A4%97-transformers). Seems like if you override `eos_token_id` you need to provide the original `self.tokenizer.tokenizer.eos_token_id`. And then we want to include our own stop tokens on top of that.

I really can't find much documentation on how to properly set stop sequences.

Reply:
Found the [documentation](https://huggingface.co/docs/transformers/v4.41.3/en/main_classes/text_generation#generation) for the parameters and updated to use `stop_strings` which seems like what we want.